File: тесты/bench10_golden_ratio.py
import time

def golden_ratio_reciprocal(n):
    x = 1.0
    i = 0
    while i < n:
        x = 1.0 + 1.0 / x
        i += 1
    return x

if __name__ == "__main__":
    t0 = time.perf_counter()
    ans = golden_ratio_reciprocal(10000000)
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000.0
    print(f"RESULT:{ans:.8f}")
    print(f"TIME_MS:{ms:.2f}")

File: тесты/bench10_golden_ratio.cpp
#include <iostream>
#include <chrono>

double golden_ratio_reciprocal(long long n) {
    double x = 1.0;
    for (long long i = 0; i < n; ++i) {
        x = 1.0 + 1.0 / x;
    }
    return x;
}

int main() {
    auto t0 = std::chrono::high_resolution_clock::now();
    double phi = golden_ratio_reciprocal(10000000);
    auto t1 = std::chrono::high_resolution_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    std::cout << "RESULT:" << phi << std::endl;
    std::cout << "TIME_MS:" << ms << std::endl;
    return 0;
}
